session03/mailroom1.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+
+# creating global list so it can be accessed from different functions
+donor_list = [('Albert White', [1254.45, 8966.70]),
+              ('Susan Smith', [9856.94, 1245.78]),
+              ('Daniel Motoko', [5897.85, 20000.48]),
+              ('Alison Dempsey', [2565.44, 500.32]),
+              ('Patrick Johnson', [6578.55, 4879, 5879.99])]
+
+
+def menu():
+    """ print menu for user to select from """
+    question = input("\nPlease select options 1, 2, or 3 from the menu "
+                         "or 'list' to view donor names: \n"
+                         "1. Send a Thank you letter\n"
+                         "2. Create a Report\n"
+                         "3. Quit\n"
+                         "What's your choice? ")
+    return question
+
+def print_donor_list():
+    """ printing 0th element from list (printing only donors' name)"""
+    print("Donor Name:\n")
+    for name in donor_list:
+        print(name[0])
+
+def find_donor(name):
+    """ finding donor name in donor_list and printing their information """
+    for donor in donor_list:
+        # case-insenstive compare
+        if name.lower() == donor[0].lower():
+            return donor
+    return None
+
+def sort_key(item):
+    """ key function used to sort the list by first (not zeroth) item"""
+    return item[1]
+
+def print_letter(name, amount):
+    """ writing a thank you letter to donor """
+    print("\nDear {},\n\nThank you for your kind donation of ${}."
+          "\n\nSincerely,\n\nA Team".format(name, amount))
+
+def print_report():
+    """ printing a report that includes donor names, contribution amounts, sum, and average """
+    report = []
+    for (name, amount) in donor_list:
```

Comment:
think a bit about naming -- "amount" is singular, but in tis case it is a list of multiple numbers -- maybe "amounts"? or "donations"?